### DISCUSSION TITLE
Time Interpolation Problem with Cross-year Months (December and January)

### DISCUSSION THREAD

**Discussion by GwenChen-NOAA** [dtcenter/METplus#2007] [Configuration] [answered]
Dear METplus team,

I'm using monthly climatology files and the time interpolation function in MET/METplus to calculate daily climatology.  Below is the setup in my Grid_Stat configuration file:

GRID_STAT_CLIMO_MEAN_FILE_NAME = woa18_decav_t12_04.nc, woa18_decav_t01_04.nc
GRID_STAT_MET_CONFIG_OVERRIDES = climo_mean = {field = [{name="t_an"; level="(0,0,*,*)";}];}
GRID_STAT_CLIMO_MEAN_REGRID_METHOD = BILIN
GRID_STAT_CLIMO_MEAN_REGRID_WIDTH = 2
GRID_STAT_CLIMO_MEAN_REGRID_VLD_THRESH = 0.5
GRID_STAT_CLIMO_MEAN_REGRID_SHAPE = SQUARE
GRID_STAT_CLIMO_MEAN_TIME_INTERP_METHOD = DW_MEAN
GRID_STAT_CLIMO_MEAN_DAY_INTERVAL = 31
GRID_STAT_CLIMO_MEAN_HOUR_INTERVAL = 6

The woa18_decav_t12_04.nc file is December climatology and the woa18_decav_t01_04.nc is January climatology.  I tried to calculate anomaly correlation coefficient for valid date 20230109, and I got the following messages in my log file:

DEBUG 3: Skipping the 19861215_000000 "t_an(0,0,*,*)" climatology field since the time offset (29376000 seconds) >= the "day_interval" entry (2678400 seconds) from file: woa18_decav_t12_04.nc
...
DEBUG 4: Found matching 19860115_000000 "t_an(0,0,*,*)" climatology field in file "woa18_decav_t01_04.nc".
DEBUG 2: Regridding the 19860115_000000 "t_an(0,0,*,*)" climatology field to the verification grid.
DEBUG 3: For forecast valid at 20230109_000000, found 1 climatology field(s) with valid time(s): 20230115_000000
DEBUG 3: Found 1 climatology fields.
DEBUG 3: Found 1 climatology mean and 0 climatology standard deviation field(s) for forecast sst(0,*,*).

It supposed to do the time interpolation to the valid date using both December and January climatology.  Below is the log messages for valid date 20221210 using November and December climatology and the same setup:

DEBUG 4: Found matching 19861115_000000 "t_an(0,0,*,*)" climatology field in file "woa18_decav_t11_04.nc".
...
DEBUG 4: Found matching 19861215_000000 "t_an(0,0,*,*)" climatology field in file "woa18_decav_t12_04.nc".
DEBUG 3: For forecast valid at 20221210_000000, found 2 climatology field(s) with valid time(s): 20221115_000000, 20221215_000000
DEBUG 3: Interpolating climatology fields at 20221115_000000 and 20221215_000000 to 20221210_000000 using the DW_MEAN interpolation method.
DEBUG 3: Found 1 climatology fields.
DEBUG 3: Found 1 climatology mean and 0 climatology standard deviation field(s) for forecast sst(0,*,*).

I think there is a bug in the MET/METplus software because the time offset failed to count backward for valid dates in early January.  Please advise.  Thank you!

**Comment by JohnHalleyGotway** [comment]
Hi Gwen, thanks for detailed explanation of the unexpected behavior you're seeing. I'll take a look at this to see if I can independently confirm the suspected problem using a different monthly climatology dataset. I'll let you know when I have something to report.

**Comment by GwenChen-NOAA** [comment]
Thanks, John!  Please also check valid dates December 16-31 and see if the time interpolation is working properly for these dates. Thanks,Gwen

**Reply by JohnHalleyGotway** [reply]
@GwenChen-NOAA, FYI, I was able to replicate the problem and understand what's going on. I'm thinking about how best to fix it and will include it in the MET-11.0.1 bugfix release. Here's the bugfix issue dtcenter/MET#2412. 

Thanks for paying close attention and identifying this problem!

**Comment by GwenChen-NOAA** [comment, marked as answer]
Thank you, John.  It's great to know a bugfix will be included in MET
11.0.1.

*Alicia*, I know MET 11.0.0 is not yet available on WCOSS2, but this fix is
essential to all RTOFS verifications with climatology.  Hopefully, we can
use MET 11.0.1 for the Beta test.

Thanks,
Gwen

**Comment by GwenChen-NOAA** [comment]
Hi Gwen,

Thanks! Yes, we hope to use MET-11.0.1 (which includes critical bug fixes
and updates) in the Beta Test and final version of EVSv1. That depends, of
course, on when MET-11.0.1 is released by DTC. If it is released too late
in our development, we can't switch to MET-11.0.1 for the Beta Test of
EVSv1. Our deadlines going forward are very rigid because of our scheduled
EVS code delivery to NCO.

Take care,
Alicia
___________________________________
*Alicia M. Bentley, Ph.D.*
Physical Scientist | Model Evaluation Group
<https://www.emc.ncep.noaa.gov/users/meg/home/>
Global Model Verification Team Lead
NOAA/NWS/NCEP/EMC (VPPPG Branch)
Work Phone: ‪(301) 683-3783‬

**Comment by GwenChen-NOAA** [comment]
Alicia,

Without this fix, anomalies and ACC calculated for valid dates Dec. 16 -
Jan. 14 are off.  We shouldn't generate an operation product with known
errors.

John, when will MET 11.0.1 be released?  If it will pass the deadline of
Beta Test (June 1), we need to let the management know about it.

Thanks,
Gwen

**Comment by GwenChen-NOAA** [comment]
Right. And as I've said in past meetings, we can't use MET-11.0.1 in EVSv1
if it's released on May 29th (for example). We would need it released
considerably earlier than that for developers to switch to it.

___________________________________
*Alicia M. Bentley, Ph.D.*
Physical Scientist | Model Evaluation Group
<https://www.emc.ncep.noaa.gov/users/meg/home/>
Global Model Verification Team Lead
NOAA/NWS/NCEP/EMC (VPPPG Branch)
Work Phone: ‪(301) 683-3783‬
